src/main/java/frc/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -0,0 +1,264 @@
+        mAngleMotor = new TrigonTalonFX(moduleConstants.angleMotorID);
+        configAngleMotor();
+
+        /* Drive Motor Config */
+        mDriveMotor = new TrigonTalonFX(moduleConstants.driveMotorID);
+        configDriveMotor();
+    }
+
+    /**
+     * Minimize the change in heading the desired swerve module state would require
+     * by potentially
+     * reversing the direction the wheel spins. Customized from WPILib's version to
+     * include placing
+     * in appropriate scope for CTRE onboard control.
+     *
+     * @param desiredState The desired state.
+     * @param currentAngle The current module angle.
+     */
+    public static SwerveModuleState optimize(
+            SwerveModuleState desiredState, Rotation2d currentAngle) {
+        double targetAngle = placeInAppropriate0To360Scope(
+                currentAngle.getDegrees(), desiredState.angle.getDegrees());
+        double targetSpeed = desiredState.speedMetersPerSecond;
+        double delta = targetAngle - currentAngle.getDegrees();
+        if(Math.abs(delta) > 90) {
+            targetSpeed = -targetSpeed;
+            targetAngle = delta > 90 ? (targetAngle -= 180) : (targetAngle += 180);
+        }
+        return new SwerveModuleState(
+                targetSpeed, Rotation2d.fromDegrees(targetAngle));
+    }
+
+    /**
+     * @param scopeReference Current Angle
+     * @param newAngle       Target Angle
+     * @return Closest angle within scope
+     */
+    private static double placeInAppropriate0To360Scope(
+            double scopeReference, double newAngle) {
+        double lowerBound;
+        double upperBound;
+        double lowerOffset = scopeReference % 360;
+        if(lowerOffset >= 0) {
+            lowerBound = scopeReference - lowerOffset;
+            upperBound = scopeReference + (360 - lowerOffset);
+        } else {
+            upperBound = scopeReference - lowerOffset;
+            lowerBound = scopeReference - (360 + lowerOffset);
+        }
+        while(newAngle < lowerBound) {
+            newAngle += 360;
+        }
+        while(newAngle > upperBound) {
+            newAngle -= 360;
+        }
+        if(newAngle - scopeReference > 180) {
+            newAngle -= 360;
+        } else if(newAngle - scopeReference < -180) {
+            newAngle += 360;
+        }
+        return newAngle;
+    }
+
+    public void setDesiredState(SwerveModuleState desiredState, boolean isOpenLoop) {
+        // Custom optimize command, since default WPILib optimize assumes
+        // continuous controller which CTRE is not
+        desiredState = optimize(
+                desiredState,
+                getState().angle
+        );
+        if(isOpenLoop) {
+            /* If we're in open loop, we don't want to use the PID controller,
+            and we just set the drive speed in percentage of the max speed */
+            double percentOutput =
+                    desiredState.speedMetersPerSecond / SwerveConstants.MAX_SPEED;
```

Comment:
don't we want the value squared during open loop? if you decide to not put it here put it somewhere